### PR TITLE
Feature/issue 177

### PR DESCRIPTION
**Pull request by rambrus**
*Issue #, if available:*
https://github.com/awslabs/deequ/issues/177

*Description of changes:*
Adding more detailed description to the success metrics *dataframe* / *json* when `where` statement is used in Analyzers, e.g.

```
 .addCheck(Check(CheckLevel.Error, "Sample check")
   .hasSize(_ > 10000).where("entity = 'A'")
   .hasSize(_ > 100000).where("entity = 'B'")
   .hasSize(_ > 100000).where("entity = 'C'")
)
```

Before this PR, the success metrics looked like:
```
+-------+--------+----+--------+
| entity|instance|name|   value|
+-------+--------+----+--------+
|Dataset|       *|Size|  5632.0|
|Dataset|       *|Size|422312.0|
|Dataset|       *|Size|126984.0|
+-------+--------+----+--------+
```

After this PR:
```
+---------+----------+----------------------------+----------+
| entity  | instance |            name            |  value   |
+---------+----------+----------------------------+----------+
| Dataset | *        | Size (where: entity = 'A') |   5632.0 |
| Dataset | *        | Size (where: entity = 'B') | 422312.0 |
| Dataset | *        | Size (where: entity = 'C') | 126984.0 |
+---------+----------+----------------------------+----------+
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala [hunk 1]
```diff
@@ -80,26 +80,56 @@ object AnalyzerContext {
       forAnalyzers: Seq[Analyzer[_, Metric[_]]])
     : Seq[SimpleMetricOutput] = {
 
-    val selectedMetrics = analyzerContext.metricMap
+    analyzerContext.metricMap
+      // Get matching analyzers
       .filterKeys(analyzer => forAnalyzers.isEmpty || forAnalyzers.contains(analyzer))
-      .values
+      // Get analyzers with successful results
+      .filter({ case (_, metrics) =>
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: no line break here please

Comment (sscdotopen) [reply src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: no line break here please

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala [hunk 2]
```diff
@@ -80,26 +80,56 @@ object AnalyzerContext {
...
+      metrics.value.isSuccess })
+      // Get metrics as Double and replace simple name with description
+      .flatMap({ case (analyzer, metrics) =>
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: please indent this

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala [hunk 3]
```diff
@@ -80,26 +80,56 @@ object AnalyzerContext {
...
+    * It helps us to show more readable success metrics
+    * (see https://github.com/awslabs/deequ/issues/177 for details)
+    *
+    * @param analyzer the Analyzer to be described
+    * @return the description of the Analyzer
+    */
+  private[this] def describeAnalyzer(analyzer: Any): String = {
+    val name = analyzer.getClass.getSimpleName
+
+    ReflectionUtils.getFieldValue(analyzer, "where") match {
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: I dont think we should do the check whether the analyzer supports filtering via reflection. This is brittle and the compiler cannot check cases where we would for example rename the field internally. Please add an extra trait to mark analyzers that have a where clause and add a method to retrieve the filter. We can pattern match on the trait then here.

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala [hunk 4]
```diff
@@ -80,26 +80,56 @@ object AnalyzerContext {
...
+      case Some(Some(e)) => s"$name (where: $e)"
+      case _ => name
+    }
+  }
+
+  private[this] case class SimpleMetricOutput( entity: String,
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: Please indent according to our code conventions https://github.com/databricks/scala-style-guide

File: src/main/scala/com/amazon/deequ/analyzers/runners/ReflectionUtils.scala [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  * permissions and limitations under the License.
+  *
+  */
+
+package com.amazon.deequ.analyzers.runners
+
+import scala.reflect.runtime.{universe => ru}
+import ru._
+
+object ReflectionUtils {
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/analyzers/runners/ReflectionUtils.scala]: Remove this.

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala (hunk 1 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: line break removed

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala (hunk 2 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: indent added

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala (hunk 4 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: indent reformatted

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala (hunk 3 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala]: Makes sense, replaced reflection with `FilterableAnalyzer` trait and added to the analyzers

File: src/main/scala/com/amazon/deequ/analyzers/runners/ReflectionUtils.scala (hunk 5 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/analyzers/runners/ReflectionUtils.scala]: done

**Comment by sscdotopen** [comment]
Thank you for your contribution!

**Pull request merged**